LoopKit/HealthKitSampleStore.swift

```diff
@@ -141,53 +153,108 @@ public class HealthKitSampleStore {
         didSet {
             // If we are now looking farther back, then reset the query
             if oldValue > observationStart {
+                log.default("observationStart changed: creating HK query")
                 createQuery()
             }
         }
     }
 
     /// The last-retreived anchor from an anchored object query
-    private var queryAnchor: HKQueryAnchor?
+    internal var queryAnchor: HKQueryAnchor? {
+        get {
+            return lockedQueryAnchor.value
+        }
+        set {
+            var changed: Bool = false
+            lockedQueryAnchor.mutate { (anchor) in
+                if anchor != newValue {
+                    anchor = newValue
+                    changed = true
+                }
+            }
+            if changed {
+                queryAnchorDidChange()
+            }
+        }
+    }
+    internal let lockedQueryAnchor: Locked<HKQueryAnchor?>
+
+    /// The last observer query completion, stored until the next anchored object query returns
+    private var observerQueryCompletionHandler: HKObserverQueryCompletionHandler? {
+        get {
+            lockedObserverQueryCompletionHandler.value
+        }
+        set {
+            var oldValue: HKObserverQueryCompletionHandler?
+            lockedObserverQueryCompletionHandler.mutate { handler in
+                oldValue = handler
+                handler = newValue
+            }
+
+            oldValue?()
+        }
+    }
+    private let lockedObserverQueryCompletionHandler: Locked<HKObserverQueryCompletionHandler?> = Locked(nil)
+
+    func queryAnchorDidChange() {
+        // Subclasses can override
```

Comment:
[nit] Consider assertion here since this should never be invoked?

Reply:
Non implementation is fine if subclasses don't want to cache query anchor.